e2e/tests/oauth/test_oauth_tokens.py

```diff
@@ -230,3 +231,144 @@ async def test_nhs_login_auth_code_flow_happy_path(
                 "NHSD-Session-URID": "ROLD-ID",
             },
         )
+
+
+@pytest.mark.asyncio
+class TestTokenExchangeTokens:
+    """Test class to confirm token exchange logic is as expected """
+    @pytest.mark.parametrize('scope', ['P9', 'P5', 'P0'])
+    async def test_nhs_login_token_exchange_access_and_refresh_tokens_generated(self, scope):
+        """
+        Ensure access token and refresh token generated by nhs login token exchange
+        Use to fetch and a new access token, refresh token pair.
+        """
+        id_token_claims = {
+            "aud": "tf_-APIM-1",
+            "id_status": "verified",
+            "token_use": "id",
+            "auth_time": 1616600683,
+            "iss": "https://internal-dev.api.service.nhs.uk",
+            "vot": "P9.Cp.Cd",
+            "exp": int(time()) + 600,
+            "iat": int(time()) - 10,
+            "vtm": "https://auth.sandpit.signin.nhs.uk/trustmark/auth.sandpit.signin.nhs.uk",
+            "jti": str(uuid4()),
+            "identity_proofing_level": scope,
+            "nhs_number": "900000000001"
+        }
+        id_token_headers = {
+            "sub": "49f470a1-cc52-49b7-beba-0f9cec937c46",
+            "aud": "APIM-1",
+            "kid": "nhs-login",
+            "iss": "https://internal-dev.api.service.nhs.uk",
+            "typ": "JWT",
+            "exp": 1616604574,
+            "iat": 1616600974,
+            "alg": "RS512",
+            "jti": str(uuid4()),
+        }
+
+        with open(ID_TOKEN_NHS_LOGIN_PRIVATE_KEY_ABSOLUTE_PATH, "r") as f:
+            contents = f.read()
+
+        client_assertion_jwt = self.oauth.create_jwt(kid="test-1")
+        id_token_jwt = self.oauth.create_id_token_jwt(
+            algorithm="RS512",
+            claims=id_token_claims,
+            headers=id_token_headers,
+            signing_key=contents,
+        )
+        resp = await self.oauth.get_token_response(
+            grant_type="token_exchange",
+            _jwt=client_assertion_jwt,
+            id_token_jwt=id_token_jwt,
+        )
+        access_token = resp['body']['access_token']
+        refresh_token = resp['body']['refresh_token']
+
+        assert bool(access_token) is True
```

Comment:
dont need bool

Reply:
Specifically, you can just call `assert access_token` here since the `assert` keyword evaluates whatever expression you give it and in Python **all values** evaluated in an expression or boolean operation are interpreted as **true** unless they are: `False`, `None`, numeric zero of all types, and empty strings/containers.